tests/unit/data_providers/test_dispatch.py

```diff
@@ -0,0 +1,111 @@
+from typing import List
+
+import pytest
+
+from app.data_providers.curation_api_client import CurationAPIFetchable
+from app.data_providers.dispatch import Dispatch
+from app.data_providers.metrics_client import MetricsFetchable
+from app.data_providers.slate_provider import SlateProvidable
+from app.data_providers.slate_provider_schemata import ExperimentSchema, SlateSchema
+from app.graphql.corpus_item import CorpusItem
+from app.models.corpus_item_model import CorpusItemModel
+from app.models.ranked_corpus_items_instance import RankedCorpusItemsInstance
+from app.rankers.algorithms import top5, top15
+
+
+class MockMetricsClient(MetricsFetchable):
+    async def get_engagement_metrics(self, ranked_items, ranker):
+        return {}
+
+class MockSlateProvider(SlateProvidable):
+    schema = SlateSchema(displayName="World's Fakest Slate", description="A selection of fake content for testing",
+                         internalDescription="We made this for our unit tests", experiments=[
+            ExperimentSchema(description="default", eligible_corpora=["CHELSEAS_AWESOME_CORPUS"], rankers=[])])
+
+    def get(self, slate_id) -> SlateSchema:
+        #offers up this slate regardless of the slate_id sent in. We test the SlateSchema logic in that class's tests.
+        return self.schema
+
+class MockCurationAPIClient(CurationAPIFetchable):
+    mock_corpus = RankedCorpusItemsInstance(
+        id="CHELSEAS_AWESOME_CORPUS",
+        description="I mean, pretty sure the id covers it ;)",
+        corpusItems=[
+            CorpusItemModel(id="https://chelseatroy.com/2022/02/10/adding-error-productions-to-the-lox-compiler/"),
```

Comment:
My understanding is that `id` in this context is a UUID. I may be wrong, if so disregard. It may be worthwhile to make the tests resemble real data for documentation.

Reply:
My understanding is that it is not. For example, the ids currently implemented for CurationToolsAPI are NEW_TAB_EN_US, NEW_TAB_EN_INTL, NEW_TAB_EN_GB, and NEW_TAB_DE_DE.